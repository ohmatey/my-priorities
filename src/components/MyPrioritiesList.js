import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import MyPriority from './MyPriority'

export const MyPrioritiesList = ({ myPriorities }) => (
  <div>
    {myPriorities.map((myPriority, i) => <MyPriority key={i} {...myPriority} />)}
  </div>
)

MyPrioritiesList.propTypes = {
  myPriorities: PropTypes.arrayOf(PropTypes.object).isRequired
}

const mapStateToProps = ({ myPriorities }) => ({
  myPriorities
})

export default connect(mapStateToProps)(MyPrioritiesList)
