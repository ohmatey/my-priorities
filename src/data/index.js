import Immutable from 'seamless-immutable'

export const prioritiesList = Immutable([
  {
    id: 0,
    name: 'Work',
    hours: 9
  },
  {
    id: 1,
    name: 'Yoga',
    hours: 1
  },
  {
    id: 2,
    name: 'Gym',
    hours: 1
  },
  {
    id: 3,
    name: 'Family Time',
    hours: 1
  }
])
