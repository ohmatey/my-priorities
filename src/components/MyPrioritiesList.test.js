import React from 'react'
import { shallowToJson } from 'enzyme-to-json'

import { MyPrioritiesList } from './MyPrioritiesList'
import { prioritiesList } from '../data/'

describe('MyPrioritiesList component', () => {
  it('matches snapshot', () => {
    const tree = shallowToJson(
      <MyPrioritiesList myPriorities={[]} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with an array', () => {
    const tree = shallowToJson(
      <MyPrioritiesList myPriorities={prioritiesList} />
    )
    expect(tree).toMatchSnapshot()
  })
})
