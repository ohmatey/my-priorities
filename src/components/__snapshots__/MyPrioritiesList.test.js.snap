// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPrioritiesList component matches snapshot 1`] = `undefined`;

exports[`MyPrioritiesList component matches snapshot with an array 1`] = `undefined`;
